modules/admin.php

```diff
@@ -1944,6 +1945,33 @@ static function _prepare_profile_modification_display($profile_modification)
         return $altered_modification;
     }
 
+    function handler_profile_json($page)
+    {
+        $res = XDB::iterator('SELECT  p.hrpid, pm.pid, pd.directory_name AS name, pm.field AS field, pm.newtext AS newtext, pm.timestamp AS timestamp
+                                FROM  profile_modifications AS pm
+                          INNER JOIN  profiles              AS p  ON (pm.pid = p.pid)
+                          INNER JOIN  profile_display       AS pd ON (pm.pid = pd.pid)
+                               WHERE  pm.type = \'self\'
+                            ORDER BY  pm.timestamp DESC, pd.directory_name');
+        $modifications = array();
+        foreach (PlIteratorUtils::foreachIterator($res) as $modif)
+        {
+            if (!array_key_exists($modif['name'], $modifications)) {
+                $modifications[$modif['name']] = array();
+            }
+            // Only keep most recent modification
+            if (!array_key_exists($display_field, $modifications[$modif['name']])) {
+                $display_field = Profile::field_display($modif['field']);
```

Comment:
PHP rightly complains because ``$display_field`` is set after being used:

    Notice: Undefined variable: display_field